include/libcyphal/transport/udp/tx_rx_sockets.hpp

```diff
@@ -96,6 +101,27 @@ class IRxSocket
     IRxSocket& operator=(const IRxSocket&)     = delete;
     IRxSocket& operator=(IRxSocket&&) noexcept = delete;
 
+    /// @brief Defines the error types that may occur during the `receive` operation.
+    ///
+    using ReceiveFailure = cetl::variant<PlatformError, ArgumentError, MemoryError>;
```

Comment:
If you don't mind I would like to start renaming several `XxxError`-s to similar but with `...Failure` suffix when it comes to a variant of errors. Idea is that it will be more clear when...
- it's a "primitive" error - use current approach with `Error` suffix
- it's a higher level variant of the above errors - use `Failure` suffix.

This is the first bird, and if we agreed on what I'm proposing then I will rename also `can::MediaError` → `can::MediaFailure`, `transport::AnyError` → `transport::AnyFailure` and couple more.

Reply:
So the proposed taxonomy is:

- Result:
  - Success
  - Failure
    - Error X
    - Error Y
    - ...

Maybe it should be documented

Reply:
Yes, something like that. I'll add todo about documenting the taxonomy, especially if it will fit well.